tests/test_modeling/test_kvcache_structural_read.py

```
@@ -0,0 +1,48 @@
+    assert torch.equal(qkv.layers[0].values, value)
+
+
+def test_genuinely_missing_attribute_still_raises():
+    real = DynamicCache()
+    real.update(torch.ones(1, 1, 1, 4), torch.ones(1, 1, 1, 4), 0)
+
+    qkv = _make_qkv_cache()
+    qkv.add_past_key_values(real)
+
+    with pytest.raises(AttributeError):
+        qkv.this_attr_does_not_exist_anywhere
```

Ruff (0.15.18)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)